src/engine-scripts/puppet/collapsedTocState.js

```diff
@@ -0,0 +1,27 @@
+const menuState = require( './menuState' );
+
+/**
+ * Open or close Vector-2022's collapsible TOC.
+ *
+ * @param {import('puppeteer').Page} page
+ * @param {string[]} hashtags
+ */
+module.exports = async ( page, hashtags ) => {
+	const isOpen = hashtags.includes( '#collapsed-toc-open' );
+	const isClosed = hashtags.includes( '#collapsed-toc-closed' );
+	const isStickyHeader = hashtags.includes( '#scroll' ) && hashtags.includes( '#logged-in' );
+
+	if ( !isOpen && !isClosed ) {
+		return;
+	}
+
+	const collapseTocButton = '.vector-toc-collapse-button';
+	await page.waitForSelector( collapseTocButton );
+	await page.evaluate( ( selector ) => {
+		const btn = document.querySelector( selector );
+		btn.click();
+	}, collapseTocButton );
```

Comment:
#collapsed-toc-open and #collapsed-toc-closed will first always click the "hide" button in order to collapse the TOC, either into the page title or sticky header. On narrow screens this will do nothing because the TOC is collapsed by default.

After the TOC is collapsed, we open or close the TOC depending on the hashtag like any other menu.